.github/workflows/check_mergability_ghstack.yml

```diff
@@ -0,0 +1,22 @@
+name: Check mergability and dependencies for ghstack prs
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  mergability-and-dependencies-check:
```

Comment:
I think that this fits better into `lint.yml` workflow.  Having a new workflow is ok, but lint seems like a natural way to keep this job

Reply:
One issue with `lint.yml` is that lint is triggered on pushes into main, which is redundant for mergeability (and in fact won't work for the workflow that requires PR number).